packages/lodestar/src/network/encoders/response.ts

```diff
@@ -62,15 +62,26 @@ export function eth2ResponseDecode(
           status = buffer.get(0);
           buffer.consume(1);
           if(status !== RpcResponseStatus.SUCCESS) {
-            logger.warn(`Received err status ${status} for method ${method}`);
+            logger.warn(`eth2ResponseDecode: Received err status ${status} for method ${method}`);
             break;
           }
         }
         if(buffer.length === 0) continue;
         if(sszLength === null) {
           sszLength = decode(buffer.slice());
+          if (decode.bytes > 10) {
+            throw new Error(`eth2ResponseDecode: Invalid number of bytes for protobuf varint ${decode.bytes}` +
+            `, method ${method}`);
+          }
           buffer.consume(decode.bytes);
         }
+        if (sszLength < type.minSize() || sszLength > type.maxSize()) {
+          throw new Error(`eth2ResponseDecode: Invalid szzLength of ${sszLength} for method ${method}`);
```

Comment:
According to the spec `From responses: ignore the response, the response MUST be considered bad server behavior.` so we have to throw Error here instead of log and break